apps/tx-builder/src/components/Builder.tsx

```diff
@@ -108,21 +130,19 @@ export const Builder = ({ contract, to }: Props): ReactElement | null => {
       const method = contract.methods[selectedMethodIndex];
 
       if (!['receive', 'fallback'].includes(method.name)) {
-        const cleanInputs = [];
-        description = method.name + ' (';
-        for (let i = 0; i < method.inputs.length; i++) {
-          const cleanValue = inputCache[i] || '';
-          cleanInputs[i] = cleanValue.charAt(0) === '[' ? JSON.parse(cleanValue.replace(/"/g, '"')) : cleanValue;
-          if (i > 0) {
-            description += ', ';
-          }
-          const input = method.inputs[i];
-          description += (input.name || input.type) + ': ' + cleanValue;
-        }
-        description += ')';
+        const parsedInputs: string[] = [];
+        const inputDescription: string[] = [];
+
+        method.inputs.forEach((input, index) => {
+          const cleanValue = inputCache[index] || '';
+          parsedInputs[index] = parseInputValue(input, cleanValue);
+          inputDescription[index] = (input.name || input.type) + ': ' + cleanValue;
+        })
+
+        description = method.name + ' (' + inputDescription.join(', ') + ')';
```

Comment:
It would be nice to use template strings instead of concatenation.

Reply:
We synced there, was adding that update in my last check :laughing: